### PR TITLE
CORE-6604: Throw 40? error for invalid inputs of the group policy file creation

### PR DESCRIPTION
**Pull request by yift-r3**

### CODE REVIEW

File: components/membership/membership-client/src/main/kotlin/net/corda/membership/client/CouldNotFindMemberException.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package net.corda.membership.client
+
+import net.corda.v5.base.exceptions.CordaRuntimeException
+
+class CouldNotFindMemberException : CordaRuntimeException("Could not find member")
```
Comment (nikinagy) [inline components/membership/membership-client/src/main/kotlin/net/corda/membership/client/CouldNotFindMemberException.kt]: Can we add some additional information: such as clues to which member we couldn't find?

File: components/membership/membership-client/src/main/kotlin/net/corda/membership/client/MemberNotAnMgmException.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package net.corda.membership.client
+
+import net.corda.v5.base.exceptions.CordaRuntimeException
+
+class MemberNotAnMgmException : CordaRuntimeException("Member is not an MGM")
```
Comment (nikinagy) [inline components/membership/membership-client/src/main/kotlin/net/corda/membership/client/MemberNotAnMgmException.kt]: Here too: which member is not an MGM?

File: components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/MGMRpcOpsImpl.kt [hunk 3]
```diff
@@ -87,7 +91,18 @@ class MGMRpcOpsImpl @Activate constructor(
...
+            return try {
+                mgmOpsClient.generateGroupPolicy(ShortHash.ofOrThrow(holdingIdentityShortHash))
+            } catch (e: CouldNotFindMemberException) {
+                throw ResourceNotFoundException("Could not find $holdingIdentityShortHash")
+            } catch (e: MemberNotAnMgmException) {
+                throw InvalidInputDataException(
+                    details = mapOf(
+                        "holdingIdentityShortHash" to holdingIdentityShortHash
+                    ),
+                    message = "Member: $holdingIdentityShortHash is not an MGM!",
```
Comment (charlieR3) [inline components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/MGMRpcOpsImpl.kt]: Nit picky suggestion:
```suggestion
                    message = "Member with holding identity $holdingIdentityShortHash is not an MGM.",
```

**Review by charlieR3 — COMMENTED** [review]
LGTM. One nit picky comment. Could be good to add the holding identity to the exceptions as Niki suggested also.

File: components/membership/membership-client/src/main/kotlin/net/corda/membership/client/MGMOpsClient.kt [hunk 4]
```diff
@@ -13,6 +14,9 @@ interface MGMOpsClient : Lifecycle {
      *
      * @param holdingIdentityShortHash The ID of the holding identity to be checked.
      * @return [String] Generated Group Policy Response.
+     * @throws [CouldNotFindMemberException] If there is not member with [holdingIdentityShortHash].
```
Comment (nikinagy) [inline components/membership/membership-client/src/main/kotlin/net/corda/membership/client/MGMOpsClient.kt]:
```suggestion
     * @throws [CouldNotFindMemberException] If there is no member with [holdingIdentityShortHash].
```

**Review by nikinagy — APPROVED** [review]
LGTM

**Review by charlieR3 — APPROVED** [review]
LGTM

**Pull request merged**